tests/AnalyticsEventReporter.test.ts

```diff
@@ -183,7 +183,56 @@ describe('Test report function', () => {
           sessionId: 'ULID1234'
         });
     });
+ it('should call post with correct fields, and use the original sessionId',
+    async () => {
+      const mockSetupSessionId = getOrSetupSessionId as jest.MockedFunction<typeof getOrSetupSessionId>;
+
+      mockSetupSessionId.mockImplementation( () => 'ULID1234');
+      mockPostWithFetch.mockRejectedValue({id: 1111, errors: ['Unauthorized request']});
+      mockUseBeacon.mockReturnValueOnce(false);
 
+      const config: AnalyticsConfig = {
+        bearer: 'bearerToken',
+        sessionTrackingEnabled: true,
+      };
+      const reporter = new AnalyticsEventReporter(config).with({
+        action: 'ADD_TO_CART',
+        referrerUrl: 'https://yext.com',
+        count: 5,
+      });
+
+      const res = await reporter.report({
+        authorization: 'Bearer shouldNotUpdate',
+        destinationUrl: 'https://google.com',
+        clientSdk: {
+          chat: '1.0.1.0',
+        },
+        sessionId: 'ULIDORIGINAL'
+      });
+
+      // Expect Unauthorized response
+      expect(res).toEqual({id: 1111, errors: ['Unauthorized request']});
+      // Expect getOrSetupSessionId to be called as sessionTrackingEnabled is set to true
```

Comment:
Adjust comment to: 

Expect getOrSetupSessionId to not be called as sessionId already provided in event payload.